{
  "affected": [
    {
      "database_specific": {
        "source": "https://storage.googleapis.com/cve-osv-conversion/osv-output/CVE-2021-29526.json",
        "vanir_signatures": [
          {
            "deprecated": false,
            "digest": {
              "function_hash": "52885584389583554662580624570638774955",
              "length": 3358.0
            },
            "id": "CVE-2021-29526-08382b02",
            "signature_type": "Function",
            "signature_version": "v1",
            "source": "https://github.com/tensorflow/tensorflow/commit/b12aa1d44352de21d1a6faaf04172d8c2508b42b",
            "target": {
              "file": "tensorflow/core/kernels/conv_ops.cc",
              "function": "ComputeConv2DDimension"
            }
          },
          {
            "deprecated": false,
            "digest": {
              "line_hashes": [
                "173687339207235999576275656733559819231",
                "121921501777671919028978630071452218039",
                "13837838493346518855615118855234985226",
                "139641609068420295264048637967276456390",
                "62924923225575467146803769758524095167",
                "88739273602309176012514572705318862526",
                "256969225563047733359944909391490043756",
                "289231885011739864568055639467248493785",
                "123914790918687186096113325014614741342",
                "162725975339914580946836678758534093446",
                "160026932477560776312510624912548423962"
              ],
              "threshold": 0.9
            },
            "id": "CVE-2021-29526-4983cc65",
            "signature_type": "Line",
            "signature_version": "v1",
            "source": "https://github.com/tensorflow/tensorflow/commit/b12aa1d44352de21d1a6faaf04172d8c2508b42b",
            "target": {
              "file": "tensorflow/core/kernels/conv_ops.cc"
            }
          }
        ]
      },
      "ranges": [
        {
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "b12aa1d44352de21d1a6faaf04172d8c2508b42b"
            }
          ],
          "repo": "https://github.com/tensorflow/tensorflow",
          "type": "GIT"
        }
      ],
      "versions": [
        "0.12.0-rc0",
        "0.12.0-rc1",
        "0.12.1",
        "0.5.0",
        "0.6.0",
        "v0.10.0",
        "v0.10.0rc0",
        "v0.11.0",
        "v0.11.0rc0",
        "v0.11.0rc1",
        "v0.11.0rc2",
        "v0.12.0",
        "v0.7.0",
        "v0.7.1",
        "v0.8.0rc0",
        "v0.9.0",
        "v0.9.0rc0",
        "v1.0.0",
        "v1.0.0-alpha",
        "v1.0.0-rc0",
        "v1.0.0-rc1",
        "v1.0.0-rc2",
        "v1.1.0",
        "v1.1.0-rc0",
        "v1.1.0-rc1",
        "v1.1.0-rc2",
        "v1.12.0",
        "v1.12.0-rc0",
        "v1.12.0-rc1",
        "v1.12.0-rc2",
        "v1.12.1",
        "v1.2.0",
        "v1.2.0-rc0",
        "v1.2.0-rc1",
        "v1.2.0-rc2",
        "v1.3.0-rc0",
        "v1.3.0-rc1",
        "v1.5.0",
        "v1.5.0-rc0",
        "v1.5.0-rc1",
        "v1.6.0",
        "v1.6.0-rc0",
        "v1.6.0-rc1",
        "v1.7.0",
        "v1.7.0-rc0",
        "v1.7.0-rc1",
        "v1.8.0",
        "v1.8.0-rc0",
        "v1.8.0-rc1",
        "v1.9.0",
        "v1.9.0-rc0",
        "v1.9.0-rc1",
        "v1.9.0-rc2"
      ]
    }
  ],
  "aliases": [
    "BIT-tensorflow-2021-29526",
    "GHSA-4vf2-4xcg-65cx",
    "PYSEC-2021-163",
    "PYSEC-2021-454",
    "PYSEC-2021-652"
  ],
  "details": "TensorFlow is an end-to-end open source platform for machine learning. An attacker can trigger a division by 0 in `tf.raw_ops.Conv2D`. This is because the implementation(https://github.com/tensorflow/tensorflow/blob/988087bd83f144af14087fe4fecee2d250d93737/tensorflow/core/kernels/conv_ops.cc#L261-L263) does a division by a quantity that is controlled by the caller. The fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow 2.4.2, TensorFlow 2.3.3, TensorFlow 2.2.3 and TensorFlow 2.1.4, as these are also affected and still in supported range.",
  "id": "CVE-2021-29526",
  "modified": "2025-10-21T06:24:33.038245Z",
  "published": "2021-05-14T20:15:11Z",
  "references": [
    {
      "type": "FIX",
      "url": "https://github.com/tensorflow/tensorflow/commit/b12aa1d44352de21d1a6faaf04172d8c2508b42b"
    },
    {
      "type": "FIX",
      "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-4vf2-4xcg-65cx"
    }
  ],
  "related": [
    "GHSA-4vf2-4xcg-65cx"
  ],
  "schema_version": "1.7.3",
  "severity": [
    {
      "score": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
      "type": "CVSS_V3"
    }
  ]
}